### PR TITLE
Prepend paths in api docs, based on $settings['dkan_api_base']

### PR DESCRIPTION
**Pull request by thierrydallacroce**
Based on a user-defined `$settings['dkan_api_base']` value (i.e. `/provider-data`) in a `settings*.php` file, prepend that value to the API paths documentation.

- [x] Test coverage exists

### CODE REVIEW

**Comment by thierrydallacroce** [comment]
@dafeder, this should be good to go. Let me know if you have any concerns or questions.
Thanks in advance.

File: modules/common/src/DkanApiDocsGenerator.php [hunk 1]
```diff
@@ -40,6 +59,10 @@ public function buildSpec(array $plugins = []) {
       $spec = array_merge_recursive($spec, $pluginSpec);
     }
```
Comment (dafeder) [inline modules/common/src/DkanApiDocsGenerator.php:61]:
```suggestion

    // Add 'dkan_api_base' setting to prefix the API path if your routing does not start
    // with your site's root URL (e.g. "data" for API paths to use "/data/api/1").
```

Comment (dafeder) [reply modules/common/src/DkanApiDocsGenerator.php:61]: @thierrydallacroce let's add a bit of explanation?

**Review by dafeder — CHANGES_REQUESTED** [review]
Looks great just one inline documentation request

**Review by dafeder — APPROVED** [review]

**Pull request merged**